pyxrf/model/fit_spectrum.py

```diff
@@ -256,17 +274,222 @@ def energy_bound_low_update(self, change):
+                    self.name_userpeak_dcenter = names[0]
+                else:
+                    self.name_userpeak_dcenter = ""
+
+                names = [name for name in self.elementinfo_list if "_delta_sigma" in name]
+                if names:
+                    self.name_userpeak_dsigma = names[0]
+                else:
+                    self.name_userpeak_dsigma = ""
+
+                names = [name for name in self.elementinfo_list if "_area" in name]
+                if names:
+                    self.name_userpeak_area = names[0]
+                else:
+                    self.name_userpeak_area = ""
+
+                if self.name_userpeak_dcenter and self.name_userpeak_dsigma:
+                    # Userpeak always has energy of 5.0 kEv, the user can set only the offset
+                    #   This is the internal representation, but we must display and let the user
+                    #   enter the true value of energy
+                    self.add_userpeak_energy = \
+                        self.param_dict[self.name_userpeak_dcenter]["value"] + 5.0
+                    # Same with FWHM for the user defined peak.
+                    #   Also, sigma must be converted to FWHM: FWHM = 2.355 * sigma
+                    self.add_userpeak_fwhm = \
+                        self._sigma_to_fwhm(self.param_dict[self.name_userpeak_dsigma]["value"]) + \
+                        self._compute_fwhm_base()
+
+                    # Create copies (before rounding)
+                    self.add_userpeak_fwhm_old = self.add_userpeak_fwhm
+
+                    # Adjust formatting (5 digits after dot is sufficient)
+                    self.add_userpeak_energy = float(f"{self.add_userpeak_energy:.5f}")
+                    self.add_userpeak_fwhm = float(f"{self.add_userpeak_fwhm:.5f}")
+
+        else:
+            raise Exception(f"Line '{eline_name}' is not in the list of selected element lines.")
+
+    def _update_userpeak_energy(self):
+
+        # According to the accepted peak model, as energy of the peak center grows,
+        #   the peak becomes wider. The most user friendly solution is to automatically
+        #   increase FWHM as the peak moves along the energy axis to the right and
+        #   decrease otherwise. So generally, the user should first place the peak
+        #   center at the desired energy, and then adjust FWHM.
+
+        # We change energy, so we will have to change FWHM as well
+        #  so before updating energy we will save the difference between
+        #  the default (base) FWHM and the displayed FWHM
+        fwhm_difference = self.add_userpeak_fwhm - self._compute_fwhm_base()
+
+        # Now we change energy.
+        energy = self.add_userpeak_energy - 5.0
+        self.param_dict[self.name_userpeak_dcenter]["value"] = energy
+        self.param_dict[self.name_userpeak_dcenter]["max"] = energy + 0.005
+        self.param_dict[self.name_userpeak_dcenter]["min"] = energy - 0.005
```

Comment:
Parametrization of `0.005` can help, as well as the explanation why this value was picked.

Reply:
Fixed

Reply:
Fixed